recipes/elfutils/all/patches/0003-provide-package-install-path-to-dlopen.patch

```diff
@@ -0,0 +1,44 @@
+--- libebl/eblopenbackend.c
++++ libebl/eblopenbackend.c
+@@ -321,20 +321,34 @@
+ 
+ 	/* Give it a try.  At least the machine type matches.  First
+            try to load the module.  */
+-	char dsoname[100];
+-	strcpy (stpcpy (stpcpy (dsoname, ORIGINDIR "libebl_"),
+-			machines[cnt].dsoname),
+-		".so");
++	char dsoname[256];
++	void *h = NULL;
++	
++	const char *elfutils_modules_env = getenv("CONAN_ELFUTILS_MODULES");
++	if (elfutils_modules_env != NULL)
++	{
++	  strcpy (stpcpy (stpcpy (stpcpy (stpcpy (dsoname, elfutils_modules_env), "/"),
++	       "libebl_"),
++	     machines[cnt].dsoname),
++	   ".so");
++	  h = dlopen (dsoname, RTLD_LAZY);
++	}
```

Comment:
@Hopobcn 
Does this solve your issue? This should remove trailing slashes.

```suggestion
+	if (elfutils_modules_env != NULL)
+	{
+    const char *dso2 = stpcpy (dsoname, elfutils_modules_env);
+    while ((*dso2 == "/") && (dso2 != dsoname)) {
+        dso2 -= 1;
+    }
+    if (dso2 != dsoname)
+    {
+	      strcpy (stpcpy (stpcpy (stpcpy (dso2, "/"), "libebl_"),
+	         machines[cnt].dsoname),
+	       ".so");
+	       h = dlopen (dsoname, RTLD_LAZY);
+    }
+}
```

Reply:
@Hopobcn 
Any update on this?
Would it be possible to increase the size of the path buffer to, let's say, 4096?

Reply:
Sorry, I haven't had enough time to look at this, Next Wednesday it's a holiday here, maybe I'll take a look.
Regarding the buffer size, I guess that the bigger the better but don't know if it could have undesired effects. I will test it on Wednesday.

Reply:
At the end I went for something similar to this:
```
const char *elfutils_modules_env = getenv("CONAN_ELFUTILS_MODULES");
if (elfutils_modules_env != NULL)
{
	 size_t len = strlen(elfutils_modules_env);
	 char* last = elfutils_modules_env + len - 1;
	 size_t destsz = (*last == '/') ? len-1 : len;
	 strcpy(dsoname, elfutils_modules_env);
	 *(dsoname+destsz) = '\0';
	 strcat(dsoname, "/libebl_");
	 strcat(dsoname, machines[cnt].dsoname);
	 strcat(dsoname, ".so");
		
	 h = dlopen(dsoname, RTLD_LAZY);
}
```
Regarding the `dsoname` I've increased the size to 512 and change it to a regular `malloc` to ease debugging with vscode. If required I can undo that change.

Reply:
Looks good!
When a recipe requires a environment variable, I try to avoid using the `CONAN_` prefix because I think conan owns it.
I want to avoid people thinking that this is a conan feature.
I would name it `ELFUTILS_CONAN_MODULES`.